applicationinsights/TelemetryClient.py

```diff
@@ -200,7 +200,18 @@ def track_request(self, name, url, success, start_time=None, duration=None, resp
         data.id = str(uuid.uuid4())
         data.name = name
         data.start_time = start_time or datetime.datetime.utcnow().isoformat() + 'Z'
-        data.duration = duration or 0
+
+        local_duration = duration or 0
+        duration_parts = []
+        for multiplier in [1000, 60, 60, 24]:
+            duration_parts.append(local_duration % multiplier)
+            local_duration = int(local_duration / multiplier)
+
+        duration_parts.reverse()
+        data.duration = ':'.join(['%02d' % a for a in duration_parts[:-1]]) + ('.%03d' % duration_parts[-1])
```

Comment:
I'd write this as `'%02d:%02d:%02d.%03d' % tuple(duration_parts)` unless you're really expecting to have to vary the number of items in `duration_parts`.

(If you don't care about Python 2.6, `'{:02d}:{:02d}:{:02d}.{:03d}'.format(*duration_parts)` is the new way of writing this.)